docs/src/test/java/docs/http/javadsl/server/testkit/MyAppWithActor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.http.javadsl.server.testkit;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Scheduler;
+import akka.actor.typed.javadsl.AskPattern;
+import akka.http.javadsl.server.AllDirectives;
+import akka.http.javadsl.server.Route;
+import akka.http.scaladsl.model.StatusCodes;
+import akka.japi.pf.PFBuilder;
+
+import java.time.Duration;
+
+public class MyAppWithActor extends AllDirectives {
+  public static class Ping {
+    public final ActorRef<String> replyTo;
+    public Ping(ActorRef<String> replyTo) {
+      this.replyTo = replyTo;
+    }
+  }
+
+    public Route createRoute(ActorRef<Ping> actorRef, Scheduler scheduler) {
+      Duration timeout = Duration.ofSeconds(3);
+      return
+        path("ping", () ->
+          onComplete(AskPattern.ask(actorRef, Ping::new, timeout, scheduler), maybeResult ->
+            maybeResult
+              .map(success -> complete("PONG!"))
+              .recover(new PFBuilder<Throwable, Route>()
+                  .matchAny(ex ->
+                          complete(StatusCodes.InternalServerError(), "An error occurred: " + ex.getMessage())
+                  ).build()
+              )
+              .get()
```

Comment:
Hm, I'd have expected this to be less imperative and relying on default route error handling? `onSuccess` rather?